api/src/pcapi/routes/shared/users.py

```diff
@@ -25,14 +26,12 @@ def get_profile():
     return serializers.SharedCurrentUserResponseModel.from_orm(user)
 
 
-# @debt api-migration
 @private_api.route("/users/token/<token>", methods=["GET"])
-def check_activation_token_exists(token):
+@spectree_serialize(on_error_statuses=[404], on_success_status=204)
+def check_activation_token_exists(token: str) -> None:
     user = users_repo.get_user_with_valid_token(token, [TokenType.RESET_PASSWORD], use_token=False)
     if user is None:
-        return jsonify(), 404
-
-    return jsonify(), 200
+        raise ResourceNotFoundError()
```

Comment:
(détail) ici je pense qu'on pourrait utiliser `flask.abort`, ça aurait le même effet :

```suggestion
        flask.abort(404)
```